test/e2e/fleetautoscaler_test.go

```diff
@@ -1040,3 +1041,270 @@ func TestCounterAutoscalerAllocated(t *testing.T) {
+		},
+		"Scale Down to MinCapacity": {
+			fas: listFas(func(fap *autoscalingv1.FleetAutoscalerPolicy) {
+				fap.List = &autoscalingv1.ListPolicy{
+					Key:         "games",
+					BufferSize:  intstr.FromInt(1),
+					MinCapacity: 10,
+					MaxCapacity: 100,
+				}
+			}),
+			wantFasErr:   false,
+			wantReplicas: 2, // Count:6 Capacity:10
+		},
+		"MinCapacity Less Than Buffer Invalid": {
+			fas: listFas(func(fap *autoscalingv1.FleetAutoscalerPolicy) {
+				fap.List = &autoscalingv1.ListPolicy{
+					Key:         "games",
+					BufferSize:  intstr.FromInt(15),
+					MinCapacity: 5,
+					MaxCapacity: 25,
+				}
+			}),
+			wantFasErr:   true,
+			wantReplicas: 3,
+		},
+		"Scale Up to Buffer": {
+			fas: listFas(func(fap *autoscalingv1.FleetAutoscalerPolicy) {
+				fap.List = &autoscalingv1.ListPolicy{
+					Key:         "games",
+					BufferSize:  intstr.FromInt(15),
+					MinCapacity: 15,
+					MaxCapacity: 100,
+				}
+			}),
+			wantFasErr:   false,
+			wantReplicas: 8, // Count:24 Capacity:40
+		},
+		"Scale Up to MaxCapacity": {
+			fas: listFas(func(fap *autoscalingv1.FleetAutoscalerPolicy) {
+				fap.List = &autoscalingv1.ListPolicy{
+					Key:         "games",
+					BufferSize:  intstr.FromInt(15),
+					MinCapacity: 15,
+					MaxCapacity: 25,
+				}
+			}),
+			wantFasErr:   false,
+			wantReplicas: 5, // Count:15 Capacity:25
+		},
+	}
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+
+			fas, err := fleetautoscalers.Create(ctx, testCase.fas, metav1.CreateOptions{})
+			if testCase.wantFasErr {
+				assert.Error(t, err)
+				return
+			}
+			assert.NoError(t, err)
+
+			log.Print("FAS: ", fas)
+
+			framework.AssertFleetCondition(t, flt, e2e.FleetReadyCount(testCase.wantReplicas))
+			fleetautoscalers.Delete(ctx, fas.ObjectMeta.Name, metav1.DeleteOptions{}) // nolint:errcheck
+
+			// Return to starting 3 replicas
+			framework.ScaleFleet(t, log, flt, 3)
+			framework.AssertFleetCondition(t, flt, e2e.FleetReadyCount(3))
+		})
+	}
+}
+
+func TestListAutoscalerAllocated(t *testing.T) {
+	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
+		t.SkipNow()
+	}
+	t.Parallel()
+
+	ctx := context.Background()
+	client := framework.AgonesClient.AgonesV1()
+	log := e2e.TestLogger(t)
+
+	defaultFlt := defaultFleet(framework.Namespace)
+	defaultFlt.Spec.Template.Spec.Lists = map[string]agonesv1.ListStatus{
+		"gamers": {
+			Values:   []string{"gamer1", "gamer2", "gamer3", "gamer4"}, // AggregateCount 12
+			Capacity: 6,                                                // AggregateCapacity 18
+		},
+	}
+
+	fleetautoscalers := framework.AgonesClient.AutoscalingV1().FleetAutoscalers(framework.Namespace)
+
+	testCases := map[string]struct {
+		fas             autoscalingv1.ListPolicy
+		wantAllocatedGs int32 // Must be >= 0 && <= 3
+		wantReadyGs     int32
+	}{
+		"Scale Down Buffer Percent": {
+			fas: autoscalingv1.ListPolicy{
+				Key:         "gamers",
+				BufferSize:  intstr.FromString("50%"),
```

Comment:
Do we also want a test for integer values?

Reply:
Mostly tested under `TestListAutoscaler` (not all cases covered). I can add some more allocated-specific tests here.

Reply:
Wouldn't mind seeing a integer based allocation BufferSize just to cover all the edges - but otherwise, this LGTM 👍🏻